packages/mls-client/src/Client.ts

```diff
@@ -0,0 +1,184 @@
+import { join } from 'node:path'
+import process from 'node:process'
+import {
+  createClient,
+  type NapiClient,
+  type NapiEncodedContent,
+} from '@xmtp/mls-client-bindings-node'
+import {
+  TextCodec,
+  type ContentCodec,
+  type ContentTypeId,
+  type EncodedContent,
+} from '@xmtp/xmtp-js'
+import { GroupUpdatedCodec } from '@/codecs/GroupUpdatedCodec'
+import { MembershipChangeCodec } from '@/codecs/MembershipChangeCodec'
+import { Conversations } from '@/Conversations'
+
+export const ApiUrls = {
+  local: 'http://localhost:5556',
+  dev: 'https://grpc.dev.xmtp.network:443',
+} as const
+
+export type XmtpEnv = keyof typeof ApiUrls
+
+/**
+ * Network options
+ */
+export type NetworkOptions = {
+  /**
+   * Specify which XMTP environment to connect to. (default: `dev`)
+   */
+  env?: XmtpEnv
+  /**
+   * apiUrl can be used to override the `env` flag and connect to a
+   * specific endpoint
+   */
+  apiUrl?: string
+}
+
+/**
+ * Encryption options
+ */
+export type EncryptionOptions = {
+  /**
+   * Encryption key to use for the local DB
+   */
+  encryptionKey?: Uint8Array | null
+}
+
+/**
+ * Storage options
+ */
+export type StorageOptions = {
+  /**
+   * Path to the local DB
+   */
+  dbPath?: string
+}
+
+export type ContentOptions = {
+  /**
+   * Allow configuring codecs for additional content types
+   */
+  codecs?: ContentCodec<any>[]
+}
+
+export type ClientOptions = NetworkOptions &
+  EncryptionOptions &
+  StorageOptions &
+  ContentOptions
+
+export class Client {
+  #innerClient: NapiClient
+  #conversations: Conversations
+  #codecs: Map<string, ContentCodec<any>>
+
+  constructor(client: NapiClient, codecs: ContentCodec<any>[]) {
+    this.#innerClient = client
+    this.#conversations = new Conversations(this, client.conversations())
+    this.#codecs = new Map(
+      codecs.map((codec) => [codec.contentType.toString(), codec])
+    )
+  }
+
+  static async create(accountAddress: string, options?: ClientOptions) {
+    const host = options?.apiUrl ?? ApiUrls[options?.env ?? 'dev']
+    const isSecure = host.startsWith('https')
+    const dbPath =
+      options?.dbPath ?? join(process.cwd(), `${accountAddress}.db3`)
+
+    return new Client(
+      await createClient(
+        host,
+        isSecure,
+        dbPath,
+        accountAddress,
+        options?.encryptionKey
+      ),
+      [
+        new MembershipChangeCodec(),
+        new GroupUpdatedCodec(),
```

Comment:
One thing to keep in mind is that `GroupUpdated` messages are illegal unless they have a `kind` of `membership_change`

Reply:
We should hide any that have a different kind at the SDK level so nobody gets bamboozled by a fake `GroupUpdated` message. I believe the mobile SDKs already do this.

Reply:
i was thinking that we should prevent manually sending `MembershipChange` and `GroupUpdated` messages. i only added the encoding to match the codec signature.

WDYT?

Reply:
I care less about blocking sends because it doesn't change the security (a motivated attacker will just fork the package and add it back).

We could have a warning or something when we detect it. But I can see having the functionality work being useful for writing tests.

Reply:
added check for invalid message in this commit: [Check for invalid group membership change messages](https://github.com/xmtp/xmtp-js/pull/613/commits/695998efec9d60e087705da9f5224a98d22f05ea)

copied [the way the iOS SDK does it](https://github.com/xmtp/xmtp-ios/blob/8f0186edc7dfd5f482725e8a3eb14c26a4173ce5/Sources/XMTPiOS/Mls/MessageV3.swift#L65-L67)